src/kOS.Safe/Compilation/KS/Compiler.cs

```diff
@@ -2141,13 +2160,76 @@ private void VisitInstructionBlock(ParseNode node)
                 PushReturnList();
             
             AddFunctionJumpVars(node);
-            VisitChildNodes(node); // nextBraceIsFunction state would get incorrectly inherited by my children here if it wasn't turned off up above. 
+            
+            int argbottomSpot = -1;
+            if (nextBraceWasFunction)
+                argbottomSpot = FindArgBottomSpot(node);
+
+            // For each child node, but interrupting for the spot
+            // where to insert the argbottom opcode:
+            for (int i = 0 ; i < node.Nodes.Count ; ++i)
+            {
+                if (i == argbottomSpot)
+                    AddOpcode(new OpcodeArgBottom());
+                
+                VisitNode(node.Nodes[i]); // nextBraceIsFunction state would get incorrectly inherited by my children here if it wasn't turned off up above.
+            }
 
             if (nextBraceWasFunction)
                 PopReturnList();
 
             EndScope(node);
         }
+        
+        /// <summary>
+        /// Checks whether or not the given ParseNode contains a declare parameter statment
+        /// inside it that pertains to this function.  Note, parameters inside functions
+        /// nested inside the current one don't count.  This method performs a recursive
+        /// walk.
+        /// </summary>
+        private bool HasParameterStmtNested(ParseNode node)
+        {
+            // Base case:
+            if (node.Token.Type == TokenType.declare_parameter_clause)
+                return true;
+
+            // Recurive case:
+            foreach (ParseNode child in node.Nodes)
+            {
+                if (child.Token.Type != TokenType.declare_function_clause) // functions nested in functions don't count
+                {
+                    if (HasParameterStmtNested(child))
+                        return true;
+                }
+            }
+                
+            return false; // default if nothing found in the above search.
+        }
+        
+        /// <summary>
+        /// When parsing a function, we need to mark the spot where the lastmost PARAMETER
+        /// statement occurred, so it knows that that's the point during runtime where it
+        /// should assert all the arguments passed in have been consumed by the function.
```

Comment:
Note, when encountering a `PARAMETER` statement inside further nested braces, as in this:
```
function foo {
    parameter a.
    if a = 1 {
        parameter b.
    }
    // more code here
}
```
The place where it should count as the end of the parameters is after it pops back out to the outer level again, meaning here:
```
function foo {
    parameter a.
    if a = 1 {
        parameter b.
    }
    // <---- put argbottom check here
    // more code here
}
```
As opposed to here:
```
function foo {
    parameter a.
    if a = 1 {
        parameter b.
        // <---- put argbottom check here
    }
    // more code here
}
```
That's important so in the case where it skips the if body, it doesn't end up also skipping the arg bottom consume.  This is designed to ensure that the argbottom consumption always occurs at the outer nest level of the function call itself, no matter how deeply nested inside the function the `PARAMETER` statement ended up being.

Because this delays the actual check for arg matching until the bottom of the brace, it is still possible for people to write function bodies that don't check parameter mismatch until too late, like so:
````
function foo {
  if true {
    parameter a,b,c.
    // rest of function body here.
  }
}
```
In that example ,the check is still deferred until getting out of the IF, because it won't insert the arg bottom check until popped back out to the nesting level of the function itself.  This doesn't break - it just devolves into working the same way it did before this pull request in that case - it can't check that you made the args match until the bottom of the function if you do this.